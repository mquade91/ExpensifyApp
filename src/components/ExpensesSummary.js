import React from 'react';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = ({expenses}) => {
  const expenseCount = expenses.length;
  const expensesTotal = numeral(getExpensesTotal(expenses)/100)
    .format('$0,0.00');
  return (
  <div>
    <h3> {`Viewing ${expenseCount} expenses totalling ${expensesTotal}`} </h3>
  </div>
  )
}

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);