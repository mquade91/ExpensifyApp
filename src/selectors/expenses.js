//GET VISIBLE EXPENSES
import moment from 'moment'

export default (expenses, {
  text,
  sortBy,
  startDate,
  endDate
}) => {
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.createdAt)

    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment) : true;

    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment) : true;

    const textMatch =
      typeof expense.description !== 'string' || expense.description.toLowerCase().includes(text.toLowerCase().replace(/\s/g, ''));

    return startDateMatch && endDateMatch && textMatch

  }).sort((a, b) => {
    if (sortBy === 'Date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else if (sortBy === 'Amount') {
      return a.amount < b.amount ? 1 : -1
    }
  })
}