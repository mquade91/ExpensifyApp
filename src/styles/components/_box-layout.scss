.box-layout {
  display: flex;
  justify-content: center;
  align-items: center;
  background: url('/images/qc.jpg');
  background-size: cover;
  height: 100vh;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(white, 0.15);
  padding: $l-size $m-size;
  border-radius: 3px;
  text-align: center;
  width: 25rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
