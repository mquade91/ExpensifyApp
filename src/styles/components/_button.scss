
.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  padding: $s-size;
  font-weight: 300;
  line-height: 1;
  font-size: $font-size-large;
  text-decoration: none;
  &--link {
    background: $dark-blue;
    color: white;
  }
  &--secondary {
    background: #888888;
  }
}
