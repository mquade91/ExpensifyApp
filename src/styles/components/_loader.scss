.loader {
  align-items: center;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  &__image {
    height: 6rem;
    width: 6rem;
  }
}