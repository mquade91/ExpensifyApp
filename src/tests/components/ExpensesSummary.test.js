import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';

import expenses from '../fixtures/expenses';

test('should render ExpenseSummary component with values', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />)
  expect(wrapper).toMatchSnapshot();
});