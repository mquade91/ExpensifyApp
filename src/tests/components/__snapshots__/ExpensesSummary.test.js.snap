// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary component with values 1`] = `
<div>
  <h3>
     
    Viewing 3 expenses totalling $1,141.95
     
  </h3>
</div>
`;
